protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1276,15 +1279,26 @@ func renderServices(services []*descriptor.Service, paths *openapiPathsObject, r
 							if err != nil {
 								return err
 							}
+
 							if len(b.PathParams) == 0 {
 								if err := schema.setRefFromFQN(meth.RequestType.FQMN(), reg); err != nil {
 									return err
 								}
 								desc = messageSchema.Description
 							} else {
-								schema = messageSchema
-								if schema.Properties == nil || len(*schema.Properties) == 0 {
-									grpclog.Warningf("created a body with 0 properties in the message, this might be unintended: %s", *meth.RequestType)
+								if messageSchema.Properties != nil && len(*messageSchema.Properties) != 0 {
+									// Combine the proto filepath and the method name for the definition's name
+									protoFilePath := lastFile.GetName()
+									protoFilePath = strings.TrimSuffix(protoFilePath, filepath.Ext(protoFilePath))
+									protoFilePath = nonAlphanumeric.ReplaceAllString(protoFilePath, "_")
+
+									defName := fmt.Sprintf("%s_%sBody", protoFilePath, meth.GetName())
```

Comment:
The file should be unique enough, but I worry that we might still have collisions between definitions in files with names that normalize to the same thing. Do we need something to keep track of already assigned names and just generate a new unique name (adding a 1 to the end or something like that)?

Reply:
I changed the strategy to use existing "fully qualified name" attributes on the service and method. This seems less clunky that parsing a filepath.